File: Algorithms/Sorting/bubble-sort.py
import random

# Linear Search Algorithm
def bubble_sort(collection):

    for i in range(len(collection)):

        for j in range(len(collection)-i-1):
            
            # Check if value at j > j+1
            if collection[j] > collection[j+1]:
                # Swapping values at j and j+1
                temp = collection[j+1]
                collection[j+1] = collection[j]
                collection[j] = temp
        
    return collection


if __name__ == "__main__":
    GREEN = '\033[92m'
    RESET = '\033[0m'
    print(f"{GREEN}Bubble Sort{RESET}")
    
    # Create random collection of items
    data = [random.randint(1, 40) for i in range(10)]

    print (data)
  
    # Call bubble sort
    result = bubble_sort(data)

    # Print result
    print(result)

File: Algorithms/Sorting/bubble-sort.c
#include<stdio.h>

void swap(double *x,double *y) {
    double temp = *x;
    *x = *y;
    *y = temp;
}

/*
   ========================
           Bubble
   ========================
*/

double* bubbleSort(double arr[], int n) {
    for (int i = 0; i < n - 1; i ++) {
        for (int j = 0; j < n - i - 1; j ++) {
            if (arr[j] > arr[j+1])
                swap(&arr[j], &arr[j+1]);

        }
    }
    return arr;
}

void print(double arr[], int n) {
    for (int i = 0; i < n; i++) {
        printf("%0.3lf ", arr[i]);
    }
    printf("\n");
}

int main() {

    double arr[] = {30, 21, 43, 23, 69, 102, 49, 34};
    int n = sizeof(arr)/sizeof(arr[0]);
    print(arr, n);
    swap(&arr[2], &arr[3]);

    double *bubble = bubbleSort(arr, n);
    print(bubble, n);

    return 0;
}
